client/context/ChildListContext.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { createContext, useContext, useReducer } from "react";
+import { getObjectChildrenUrl } from "../util/routes";
+import { useFetchContext } from "./FetchContext";
+
+interface Children {
+    numFound?: number;
+    start?: number;
+    docs?: Record<string, string>[];
+}
+
+interface ChildListState {
+    childStorage: Record<string, Children>;
+}
+
+const childListContextParams: ChildListState = {
+    childStorage: {},
```

Comment:
Could we use a typescript Map here? https://howtodoinjava.com/typescript/maps/

Reply:
That does seem like a very good fit. My only question is how should we handle this in the reducer? Do we need to be concerned about mutability? That is, can we just mutate the Map in the state and then return the state, or is it strictly necessary to create a fresh copy of the whole state? (I've been assuming the latter to be the case, but maybe I'm wrong).

Reply:
(After further discussion outside of this PR, we decided not to pursue this due to mutability concerns; I'm marking this conversation as resolved).